streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -61,117 +59,48 @@
-            // completed yet; do not start the first transaction until the topology has been initialized later
-            try {
-                producer.initTransactions();
-            } catch (final TimeoutException exception) {
-                final String errorMessage = "Timeout exception caught when initializing transactions for task " + taskId + ". " +
-                    "\nThe broker is either slow or in bad state (like not having enough replicas) in responding to the request, " +
-                    "or the connection to broker was interrupted sending the request or receiving the response. " +
-                    "\n Consider overwriting `max.block.ms` to a larger value to avoid timeout errors";
-
-                // TODO K9113: we do NOT try to handle timeout exception here but throw it as a fatal error
-                throw new StreamsException(errorMessage, exception);
-            } catch (final KafkaException exception) {
-                throw new StreamsException("Error encountered while initializing transactions for task " + taskId, exception);
-            }
-        }
-    }
-
-    private void maybeBeginTxn() {
-        if (eosEnabled && !transactionInFlight) {
-            try {
-                producer.beginTransaction();
-            } catch (final ProducerFencedException error) {
-                throw new TaskMigratedException(taskId, "Producer get fenced trying to begin a new transaction", error);
-            } catch (final KafkaException error) {
-                throw new StreamsException("Producer encounter unexpected error trying to begin a new transaction", error);
-            }
-            transactionInFlight = true;
-        }
-    }
-
-    private void maybeAbortTxn() {
-        if (eosEnabled && transactionInFlight) {
-            try {
-                producer.abortTransaction();
-            } catch (final ProducerFencedException ignore) {
-                /* TODO
-                 * this should actually never happen atm as we guard the call to #abortTransaction
-                 * -> the reason for the guard is a "bug" in the Producer -- it throws IllegalStateException
-                 * instead of ProducerFencedException atm. We can remove the isZombie flag after KAFKA-5604 got
-                 * fixed and fall-back to this catch-and-swallow code
-                 */
-
-                // can be ignored: transaction got already aborted by brokers/transactional-coordinator if this happens
-            } catch (final KafkaException error) {
-                throw new StreamsException("Producer encounter unexpected error trying to abort the transaction", error);
-            }
-            transactionInFlight = false;
-        }
+        this.offsets = new HashMap<>();
     }
 
     public void commit(final Map<TopicPartition, OffsetAndMetadata> offsets) {
         if (eosEnabled) {
-            maybeBeginTxn();
```

Comment:
Note that we don't even need to call a "maybeInitTx" on `StreamsProducer` -- it embeds the corresponding logic internally and fully automatic.